symengine/expression.h

```diff
@@ -120,11 +120,22 @@ class Expression
     {
         return eq(*m_basic, *other.m_basic);
     }
+
+    bool operator<(const Expression &other) const
+    {
+        return eq(*m_basic, *other.m_basic);
+    }
+     
     //! Overload check not equal (!=)
     bool operator!=(const Expression &other) const
     {
         return not (*this == other);
     }
+    //! Overload (>)
+    bool operator>(const Expression &other) const
```

Comment:
I'd avoid defining these as `__cmp__` is not mathematical comparison, but structural comparison.